osu.Game.Rulesets.Mania/UI/ManiaPlayfield.cs

```diff
@@ -1,153 +1,103 @@
-using osu.Framework.Allocation;
-using System.Linq;
 using System.Collections.Generic;
+using System.Linq;
 using osu.Framework.Configuration;
-using osu.Game.Rulesets.Objects.Drawables;
-using osu.Game.Rulesets.Mania.Objects.Drawables;
-using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
 using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Mania.Beatmaps;
+using osu.Game.Rulesets.Mania.Objects;
+using osu.Game.Rulesets.Mania.Objects.Drawables;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.UI;
+using OpenTK;
 
 namespace osu.Game.Rulesets.Mania.UI
 {
     public class ManiaPlayfield : ScrollingPlayfield
     {
-        public const float HIT_TARGET_POSITION = 50;
+        /// <summary>
+        /// list mania column stages
+        /// </summary>
+        private readonly FillFlowContainer<ManiaColumnStage> listColumnStages;
+
+        /// <summary>
+        /// Whether this playfield should be inverted. This flips everything inside the playfield.
+        /// </summary>
+        public readonly Bindable<bool> Inverted = new Bindable<bool>(true);
 
-        private SpecialColumnPosition specialColumnPosition;
         /// <summary>
         /// The style to use for the special column.
         /// </summary>
         public SpecialColumnPosition SpecialColumnPosition
         {
-            get { return specialColumnPosition; }
+            get => listColumnStages.FirstOrDefault()?.SpecialColumnPosition ?? SpecialColumnPosition.Normal;
             set
             {
-                if (IsLoaded)
-                    throw new InvalidOperationException($"Setting {nameof(SpecialColumnPosition)} after the playfield is loaded requires re-creating the playfield.");
-                specialColumnPosition = value;
+                foreach (var singleStage in listColumnStages)
+                {
+                    singleStage.SpecialColumnPosition = value;
+                }
             }
         }
 
-        /// <summary>
-        /// Whether this playfield should be inverted. This flips everything inside the playfield.
-        /// </summary>
-        public readonly Bindable<bool> Inverted = new Bindable<bool>(true);
-
-        private readonly FlowContainer<Column> columns;
-        public IEnumerable<Column> Columns => columns.Children;
-
-        protected override Container<Drawable> Content => content;
-        private readonly Container<Drawable> content;
-
-        private List<Color4> normalColumnColours = new List<Color4>();
-        private Color4 specialColumnColour;
-
-        private readonly Container<DrawableManiaJudgement> judgements;
-
-        private readonly int columnCount;
+        public List<Column> Columns
+        {
+            get
+            {
+                var list = new List<Column>();
+                foreach (var stage in listColumnStages)
+                {
+                    list.AddRange(stage.Columns);

```

Comment:
Could this be simplified to `get => listColumnStages.Select(c => c.Columns)`?

Reply:
I tried it and it should use 
public List<Column> Columns => listColumnStages.SelectMany(x => x.Columns).ToList();
instead